### PR TITLE
[1.8.9] Base code for fixing all machine types

### PR DESCRIPTION
**Pull request by AlexIIL**
[ [api](https://github.com/CovertJaguar/Railcraft-API/pull/12) | main ]

The main change this makes is that ````BlockMachine````, ````IEnumMachine```` and ````IMachineProxy```` are generic, as this makes using block state properties a bit more sane than without.

Other than that this is all just fixes, and a few more helper functions.

### CODE REVIEW

File: src/main/java/mods/railcraft/common/blocks/machine/alpha/TileRollingMachine.java [hunk 1]
```diff
@@ -330,8 +330,34 @@ public boolean hasCustomName() {
...
+    public int getFieldCount() {
+        return 0;
+    }
+
+    @Override
+    public void setField(int id, int value) {}
+
+    @Override
+    public ItemStack removeStackFromSlot(int index) {
+        ItemStack stack = getStackInSlot(index);
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/machine/alpha/TileRollingMachine.java]: return inv.removeStackFromSlot(index)?

File: src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java [hunk 2]
```diff
@@ -114,7 +114,9 @@ public TileSteamOven() {
     public static void placeSteamOven(World world, BlockPos pos, List<ItemStack> input, List<ItemStack> output) {
         MultiBlockPattern pattern = TileSteamOven.patterns.get(0);
         Map<Character, IBlockState> blockMapping = new HashMap<Character, IBlockState>();
-        blockMapping.put('B', EnumMachineAlpha.STEAM_OVEN.ordinal());
+        IBlockState base = RailcraftBlocks.getBlockMachineAlpha().getDefaultState();
+        IBlockState steamOven = base.withProperty(MachineProxyAlpha.VARIANT, EnumMachineAlpha.STEAM_OVEN);
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java]: EnumMachineAlpha.STEAM_OVEN.getState()?

File: src/main/java/mods/railcraft/common/blocks/machine/alpha/TileTradeStation.java [hunk 3]
```diff
@@ -97,14 +99,20 @@ public void update() {
...
         attemptTrade(villagers, 1);
         attemptTrade(villagers, 2);
     }
 
     private void modifyNearbyAI() {
-        List<EntityVillager> villagers = MiscTools.getNearbyEntities(worldObj, EntityVillager.class, xCoord, yCoord - 1, yCoord + 3, zCoord, 20);
+        float x = getPos().getX();
+        float y = getPos().getY();
+        float z = getPos().getZ();
+        List<EntityVillager> villagers = MiscTools.getNearbyEntities(worldObj, EntityVillager.class, x, y - 1, y + 3, z, 20);
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/machine/alpha/TileTradeStation.java]: TODO: Make into function?

File: src/main/java/mods/railcraft/common/blocks/machine/beta/TileEngineSteamHobby.java [hunk 4]
```diff
@@ -218,6 +217,33 @@ public boolean canExtractItem(int index, ItemStack stack, EnumFacing direction)
     }
 
     @Override
+    public ItemStack removeStackFromSlot(int index) {
+        ItemStack stack = getStackInSlot(index);
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/machine/beta/TileEngineSteamHobby.java]: return inv.removeStackFromSlot(index)?

**Comment by CovertJaguar** [comment]
Looks pretty good, just some minor issues.

File: src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java (hunk 2 above)
Comment (AlexIIL) [inline src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java]: That does sound like a helpful helper function

File: src/main/java/mods/railcraft/common/blocks/machine/beta/TileEngineSteamHobby.java (hunk 4 above)
Comment (AlexIIL) [inline src/main/java/mods/railcraft/common/blocks/machine/beta/TileEngineSteamHobby.java]: Both of those were copy/pasted, a side effect of suddenly having even more methods to support.

**Pull request merged**